src/uct/base/uct_iface.h

```diff
@@ -256,6 +256,21 @@ enum {
 UCS_CLASS_DECLARE(uct_iface_h, uct_iface_ops_t, uct_md_h);
 UCS_CLASS_DECLARE(uct_ep_t, uct_iface_h);
 
+/**
+ * 
+ */
+#define UCT_IFACE_FILL_INFO_STR_BUF(params, fmt, ...) do { \
+    char *info_str_buf      = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, \
+                                              params, info_string, INFO_STRING, \
+                                              NULL); \
+    size_t info_str_buf_len = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, \
+                                              params, info_string_length, \
+                                              INFO_STRING_LENGTH, 0); \
+    if ((info_str_buf != NULL) && (info_str_buf[0] == '\0')) { \
```

Comment:
why need to check info_str_buf[0] ?

Reply:
If this routine is called more than once in a flow, it'd probably be a case where the first call is in the deepest level of the tree and the second call is a more generic description (for cases that no specific description is provided), and if so then we would like to choose the first description. But after a second look at the code I don't find such cases so I'll remove it

Reply:
thanks, in general the call should ignore whatever was in info_str_buf before and overwrite it

Reply:
This is right as long as info_str_buf is not called more than once in nested calls, so if we go this way we need to make sure of it in every flow